Assets/Plugins/Editor/Uplift/Schemas/Repositories/GithubRepository.cs

```diff
@@ -84,10 +85,22 @@ public override Upset[] ListPackages()
                     "Parsing " + asset.name, 
                     (float)(index++) / upsetAssets.Length
                 );
-                
-                using(StreamReader sr = new StreamReader(GitHub.GetAssetStream(asset, GetToken())))
+
+                if (!TryGetCachedItem(asset.name, out assetPath))
+                {
+                    using (StreamReader sr = new StreamReader(GitHub.GetAssetStream(asset, GetToken())))
```

Comment:
I would suggest moving this code to a Util funciton something like

````
public static void CopyStream(Stream input, Stream output, int buffer_size = 0x8000)
{
    byte[] buffer = new byte[32768];
    int read;
    while ((read = input.Read(buffer, 0, buffer.Length)) > 0)
    {
        output.Write (buffer, 0, read);
    }
}
````

Note that I used 32k instead of 4k by default.

Reply:
And then maybe create a `CacheAsset` method

Reply:
Noted for the stream copying, I just remembered that we have a stream extension to copy streams, so I used that.

Regarding the cache; if that's okay with you, I'd rather spend some time implement a proper caching system, which would allow us to cache other things than just assets. Creating a method for this would take us in the wrong direction. I 100% agree that it should be extracted, but to something a bit nicer imo. It is not perfect as it is, but it works :)

Reply:
A method in my mind was to move it close to the other `GetCachePath` cache related methods. Do as you feel is best. It's not a big deal.

Great that we reuse the extension method.